Sources/SwiftFusion/Core/LieGroup.swift

```diff
@@ -1,17 +1,172 @@
   @differentiable(wrt: (lhs, rhs))
-  static func * (_ lhs: Self, _ rhs: Self) -> Self
-  
+  @differentiable(wrt: lhs)
+  @differentiable(wrt: rhs)
+  public static func * (_ lhs: Self, _ rhs: Self) -> Self {
+    return Self(coordinate: lhs.coordinate * rhs.coordinate)
+  }
+
+  /// Derivative of `*` with respect to both sides.
+  ///
+  /// We define separate derivatives with respect to different sets of arguments so that we can
+  /// avoid doing unnecessary work when some of the arguments are not changing.
+  ///
+  /// Swift AD can compute this, but we know a mathematical expression for the derivative that
+  /// is more efficient than the compiler can figure out.
+  @derivative(of: *, wrt: (lhs, rhs))
+  @usableFromInline
+  static func vjpGroupOperationWrtBoth(_ lhs: Self, _ rhs: Self) ->
+    (value: Self, pullback: (TangentVector) -> (TangentVector, TangentVector))
+  {
+    // Derivative from https://github.com/borglab/gtsam/blob/develop/doc/math.pdf sections 5.2 and
+    // 5.4. We use the transpose of the Adjoint because the pullback is the transpose of the
+    // differential.
+    return (lhs * rhs, { (rhs.inverse().AdjointTranspose($0), $0) })
+  }
+
+  /// Derivative of `*` with respect to the left side.
+  ///
+  /// We define separate derivatives with respect to different sets of arguments so that we can
+  /// avoid doing unnecessary work when some of the arguments are not changing.
+  ///
+  /// Swift AD can compute this, but we know a mathematical expression for the derivative that
+  /// is more efficient than the compiler can figure out.
+  @derivative(of: *, wrt: lhs)
+  @usableFromInline
+  static func vjpGroupOperationWrtLhs(_ lhs: Self, _ rhs: Self) ->
+    (value: Self, pullback: (TangentVector) -> TangentVector)
+  {
+    // Derivative from https://github.com/borglab/gtsam/blob/develop/doc/math.pdf section 5.4. We
+    // use the transpose of the Adjoint because the pullback is the transpose of the differential.
+    return (lhs * rhs, { rhs.inverse().AdjointTranspose($0) })
+  }
+
+  /// Derivative of `*` with respect to the right side.
+  ///
+  /// We define separate derivatives with respect to different sets of arguments so that we can
+  /// avoid doing unnecessary work when some of the arguments are not changing.
+  ///
+  /// Swift AD can compute this, but we know a mathematical expression for the derivative that
+  /// is more efficient than the compiler can figure out.
+  @derivative(of: *, wrt: rhs)
+  @usableFromInline
+  static func vjpGroupOperationWrtRhs(_ lhs: Self, _ rhs: Self) ->
+    (value: Self, pullback: (TangentVector) -> TangentVector)
+  {
+    // Derivative from https://github.com/borglab/gtsam/blob/develop/doc/math.pdf section 5.2. We
+    // use the transpose of the Adjoint because the pullback is the transpose of the differential.
+    return (lhs * rhs, { $0 })
+  }
+
+  /// The Adjoint group action of `self` on `v`.
+  public func Adjoint(_ v: TangentVector) -> TangentVector {
```

Comment:
I think this is against the "conventional" way of naming functions in Swift, right? But considering the math background I think this one is better